sdk/core/core/src/az_client_secret_credential.c

```diff
@@ -43,125 +43,75 @@ static AZ_NODISCARD az_result az_token_credential_get_token(
     az_span * const out_result) {
   AZ_CONTRACT_ARG_NOT_NULL(credential);
   AZ_CONTRACT_ARG_NOT_NULL(out_result);
-  AZ_CONTRACT_ARG_VALID_SPAN(resource_url);
-
-  AZ_CONTRACT_ARG_VALID_MUT_SPAN(response_buf);
-  AZ_CONTRACT_ARG_VALID_SPAN(credential->tenant_id);
-
-  AZ_CONTRACT_ARG_VALID_SPAN(credential->client_id);
-  AZ_CONTRACT_ARG_VALID_SPAN(credential->client_secret);
-
   {
-    AZ_CONTRACT(resource_url.size >= 12, AZ_ERROR_ARG);
-    AZ_CONTRACT(credential->tenant_id.size >= 32, AZ_ERROR_ARG);
-    AZ_CONTRACT(credential->client_id.size >= 32, AZ_ERROR_ARG);
-    AZ_CONTRACT(credential->client_secret.size >= 32, AZ_ERROR_ARG);
-  }
-
-  static az_span const auth_url1 = AZ_CONST_STR("https://login.microsoftonline.com/");
-  static az_span const auth_url2 = AZ_CONST_STR("/oauth2/token");
-
-  static az_span const auth_body1 = AZ_CONST_STR("grant_type=client_credentials&client_id=");
-  static az_span const auth_body2 = AZ_CONST_STR("&client_secret=");
-  static az_span const auth_body3 = AZ_CONST_STR("&resource=");
+    uint8_t auth_url_buf[100] = { 0 };
+    az_span auth_url = { 0 };
+    {
+      static az_span const auth_url1 = AZ_CONST_STR("https://login.microsoftonline.com/");
```

Comment:
Use more descriptive names for url1 and url2.  These are also never going to change and could be defined constants
url1 is the protocol + domain name
url2 is the path

Reply:
You end the domain with '/' and start path with '/'.  That results in '//' is the url you craft.

Reply:
There is AZ_RETURN_IF_FAILED(az_uri_encode(credential->tenant_id, &builder)); between the two calls, so it is "domain/" + "guid" + "/path". Now that I rewritten the code it is even more clear.